walletkit: add  SubmitPackage

bhandras opened the pull request:
This is experimental and is pending on https://github.com/lightningnetwork/lnd/pull/9784

#### Pull Request Checklist

- [x] PR is opened against correct version branch.
- [ ] Version compatibility matrix in the README and minimal required version
      in `lnd_services.go` are updated.
- [ ] Update `macaroon_recipes.go` if your PR adds a new method that is called
  differently than the RPC method it invokes.

bhandras committed:
walletkit: add SubmitPackage

guggero commented:
BTW, just as an info: Each sub client of the `lndclient` now has a method `RawClientWithMacAuth` which allows you to call any underlying RPC method without needing to add it to the code here first. Could be useful during development.

hieblmi committed:
closechannel: add functional options

hieblmi committed:
misc: fix formatting and typos

guggero committed:
Merge pull request #224 from hieblmi/no-wait

closechannel: add functional options

guggero committed:
Merge branch 'lnd-19-0'

guggero committed:
signer_client: correctly use 32-byte key for schnorr

This was incorrect all along, seems like we haven't actually used the
VerifySchnorr option in any project yet.

guggero committed:
Merge pull request #225 from lightninglabs/verifymessage-schnorr

signer_client: correctly use 32-byte key for schnorr

bhandras committed:
signer: allow passing nil keyLocator to DeriveSharedKey

Previously if the `keyLocaltor` argument was unset `DeriveSharedKey`
would crash.

bhandras committed:
Merge pull request #226 from lightninglabs/derivesharedkey-empty-keylocator

signer: allow passing nil keyLocator to DeriveSharedKey

starius committed:
walletkit: add RPC GetTransaction

https://lightning.engineering/api-docs/api/lnd/wallet-kit/get-transaction/

The RPC is available since LND 0.18.0.

guggero committed:
Merge pull request #227 from starius/get-transaction

walletkit: add RPC GetTransaction

dependabot[bot] committed:
build(deps): bump github.com/go-viper/mapstructure/v2

---
updated-dependencies:
- dependency-name: github.com/go-viper/mapstructure/v2
  dependency-version: 2.3.0
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

bhandras committed:
lndclient: make notifier options exported (to help mocking)

starius committed:
GetInfo: add more fields to response

Add fields: CommitHash, BestBlockHash, Color, NumPeers.

guggero committed:
Merge pull request #228 from lightninglabs/dependabot/go_modules/github.com/go-viper/mapstructure/v2-2.3.0

build(deps): bump github.com/go-viper/mapstructure/v2 from 2.2.1 to 2.3.0

dependabot[bot] committed:
build(deps): bump github.com/go-viper/mapstructure/v2 in /tools

Bumps [github.com/go-viper/mapstructure/v2](https://github.com/go-viper/mapstructure) from 2.2.1 to 2.3.0.
- [Release notes](https://github.com/go-viper/mapstructure/releases)
- [Changelog](https://github.com/go-viper/mapstructure/blob/main/CHANGELOG.md)
- [Commits](https://github.com/go-viper/mapstructure/compare/v2.2.1...v2.3.0)

---
updated-dependencies:
- dependency-name: github.com/go-viper/mapstructure/v2
  dependency-version: 2.3.0
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

guggero committed:
Merge pull request #230 from lightninglabs/dependabot/go_modules/tools/github.com/go-viper/mapstructure/v2-2.3.0

build(deps): bump github.com/go-viper/mapstructure/v2 from 2.2.1 to 2.3.0 in /tools

guggero committed:
Merge pull request #229 from starius/getinfo-more-fields

GetInfo: add more fields to response

guggero committed:
Merge pull request #231 from lightninglabs/exported-notifier-options

lndclient: make notifier options exported (to help mocking)

starius committed:
RegisterSpendNtfn: support reorg channel

The corresponding API of LND already supports notifying the caller about
a reorg. I reused NotifierOption's used for RegisterConfirmationsNtfn already
to pass a channel notified upon a reorg affecting a reported confirmation.

starius committed:
listen for ctx expiry when sending spend and conf

Avoid a deadlock in case the caller doesn't consume notifications about spending
or confirmations after cancelling a call.

guggero committed:
Merge pull request #233 from starius/spend-reorg

RegisterSpendNtfn: support reorg channel

bhandras commented:
Closing this as @ellemouton took over in https://github.com/lightninglabs/lndclient/pull/280
